docs/mech-tool/index.md

```diff
@@ -346,8 +386,70 @@ bash run_service.sh
 docker logs mech_abci_0 --follow
 ```
 
-**5.** Stop the mech service: 
+**5.** You can send a request, by changing the value of `priority_mech_address` in the dictionary of the chain chosen when setting up the RPC endpoint in `.mech_client/configs/mechs.json` by the address of your Mech. This address can be found in `.mech_quickstart/local_config.json`, key `mech_address`. Then use the mechx command: 
+
+```
+mechx interact <prompt> --tool <tool_name> --chain-config <chain>
+```
+
+where `<prompt>` is replaced by the chosen prompt and `<tool_name>` by the name of your tool, and `<chain>` is replaced by the 
+name of the chosen chain.
+
+**6.** Stop the mech service: 
 
 ```
 ./stop_service.sh
 ```
+
+## 5. Registering an agent on the Mech Marketplace
+
+In case you have already a Mech service deployed on Olas Registry and want to put it to work for other agents, you only need to register it on the Mech Marketplace. 
+
+In order to do so, follow the instructions below.
+
+**1.** Find [there](https://github.com/valory-xyz/ai-registry-mech/blob/main/docs/configuration.json) the address of MechMarketPlaceProxy for the chosen network.
+
+**2.** Trigger the function `create` of this contract with the following inputs (in order):
+
+- The service ID.
+- The Mech Factory address for the selected network and payment model. To find the correct address, refer to the [configuration file](https://github.com/valory-xyz/ai-registry-mech/blob/main/docs/configuration.json). Search for the address that matches the chosen payment model:
+
+    - For Native, look for the MechFactoryFixedPriceNative address.
+
+    - For Token: MechFactoryFixedPriceToken
+
+    - For Nevermined, find MechFactoryNvmSubscriptionNative.
+
+The parameter payload corresponds to the price of the Mech (also called maxDeliveryRate). In order to convert it in the format uint256 you can proceed as follows:
+
+- Convert to Wei: e.g. 1 xDAI  = 10^18 wei.
+- Use [ABI Hashex Encoder](https://abi.hashex.org/).
+- Select uint256 as the type and enter the value 1000000000000000000 (which represents 1 xDAI in wei).
+- The tool will generate the encoded result: 0000000000000000000000000000000000000000000000000de0b6b3a7640000
+
+You can find a script for triggering this function [there](https://github.com/Sfgangloff/ai-registry-mech/tree/main/scripts/mech_registration) for each payment model. Clone the repository: 
+
+```
+git clone https://github.com/Sfgangloff/ai-registry-mech.git
+```
+
+Choose the one which corresponds to the chosen payment model, and replace the name of the network on line 6. Then add your private key, serviceId and maximum price (payload) in the globals file which corresponds to the chosen network. Finally, run the script. For instance, for a native fixed price Mech: 
```

Comment:
since we have the script, we should also convert the maxDelivery price and get rid of all the instructions to get the encoded payload